github-auth-provider/main.go

```diff
@@ -133,13 +135,14 @@ func main() {
 	})
 	mux.HandleFunc("/obot-get-state", getState(oauthProxy))
 	mux.HandleFunc("/obot-get-user-info", func(w http.ResponseWriter, r *http.Request) {
-		userInfo, err := profile.FetchGitHubProfile(r.Context(), r.Header.Get("Authorization"), "https://api.github.com/user")
+		userInfo, err := profile.FetchUserProfile(r.Context(), r.Header.Get("Authorization"))
 		if err != nil {
 			http.Error(w, fmt.Sprintf("failed to fetch user info: %v", err), http.StatusBadRequest)
 			return
 		}
 		json.NewEncoder(w).Encode(userInfo)
 	})
+	mux.HandleFunc("/obot-list-auth-groups", listGroups(legacyOpts.LegacyProvider.GitHubToken))
```

Comment:
Do we ever pass the GitHubToken into auth provider? i thought it is already coming from the obot where it fetch from state.